apps/controllers/ControllerAdminBase.php

```diff
@@ -136,6 +144,73 @@ public function initialize()
+        =            Set menu struct data            =
+        ============================================*/
+        
+        /**
+         *
+         * In every menu (include sub-menu), we must have 3 element in array:
+         * @param code : name of controller, action assigned in controller
+         * @param name : name to show to user
+         * @param sub  : sub menu. let empty array if this menu hasn't sub menu
+         *
+         */
+        
+
+        $this->menuStruct[] = array("code" => "pages",
+                                    "name" => "Pages",
+                                    "sub" => []
+                                    );
+        $this->menuStruct[] = array("code" => "users",
+                                    "name" => "Users",
+                                    "sub" => []
+                                    );
+        $this->menuStruct[] = array("code" => "posts",
+                                    "name" => "Posts",
+                                    "sub" => 
+                                        [
+                                            [
+                                                "code"  =>  "sticky",
+                                                "name"  =>  "Sticky posts",
+                                                "sub"   =>  []
+                                            ]
+                                        ]
+                                    );
+        $this->menuStruct[] = array("code" => "template",
+                                    "name" => "Templates",
+                                    "sub" => []
+                                    );
+        $this->menuStruct[] = array("code" => "tags",
+                                    "name" => "Tags",
+                                    "sub" => []
+                                    );
+        $this->menuStruct[] = array("code" => "settings",
+                                    "name" => "Settings",
+                                    "sub" => 
+                                        [
+                                            [
+                                                "code" => "general",
+                                                "name" => "General",
+                                                "sub"  => []
+                                            ],
+                                            [
+                                                "code" => "logo",
+                                                "name" => "Change Logo",
+                                                "sub"  => []
+                                            ]
+                                        ]
+                                    );
+        $this->menuStruct[] = array("code" => "update",
+                                    "name" => "Updates",
+                                    "sub" => array()
+                                    );
```

Comment:
If so, we can store this structure into config or db menu table and use somethig like this:
```php
$di->setShared('menuStruct', function() use ($config) {
    // if structure received from db table instead getting from $config
    // we need to store it to cache for reducing db connections
    $struct = $config->get('menuStruct')->toArray();

    return $struct;
});
```